src/graph/graph.c

```diff
@@ -480,27 +550,62 @@ int Graph_DeleteEdge(Graph *g, Edge *e) {
     GrB_Matrix_extractElement_BOOL(&x, M, dest_id, src_id);
     if(!x) return 0;
 
-    res = GxB_Matrix_Delete(M, dest_id, src_id);
-    assert(res == GrB_SUCCESS);
-
-    M = _Graph_GetRelationMap(g, r);
-    res = GxB_Matrix_Delete(M, dest_id, src_id);
-    assert(res == GrB_SUCCESS);
+    EdgeID edge_id;
+    GrB_Matrix R = _Graph_GetRelationMap(g, r);
+    GrB_Matrix_extractElement_UINT64(&edge_id, R, dest_id, src_id);
 
-    // See if source is connected to destination with additional edges.
-    bool connected = false;
-    int relationCount = Graph_RelationTypeCount(g);
-    for(int i = 0; i < relationCount; i++) {
-        M = Graph_GetRelationMatrix(g, i);
-        GrB_Matrix_extractElement_BOOL(&connected, M, dest_id, src_id);
-        if(connected) break;
-    }
+    if(SINGLE_EDGE(edge_id)) {
+        /* Single edge of type R connecting src to dest.
+         * delete entry from both M and R. */
+        assert(GxB_Matrix_Delete(M, dest_id, src_id) == GrB_SUCCESS);        
+        assert(GxB_Matrix_Delete(R, dest_id, src_id )== GrB_SUCCESS);
+    
+        // See if source is connected to destination with additional edges.
+        bool connected = false;
+        int relationCount = Graph_RelationTypeCount(g);
+        for(int i = 0; i < relationCount; i++) {
+            if(i == r) continue;
+            M = Graph_GetRelationMatrix(g, i);
+            info = GrB_Matrix_extractElement_BOOL(&connected, M, dest_id, src_id);
+            if(info == GrB_SUCCESS) break;
```

Comment:
I'm guessing that this works because GraphBLAS will report a failure unless there was an element to extract, and that element (`connected`) will always be true. I think the behavior would be more obvious if we wrote the condition more explicitly, though: `if (connected && info == GrB_SUCCESS)` or similar.

Reply:
`GrB_Matrix_extractElement_BOOL` will return `GrB_SUCCESS` if there's a value at position: [dest_id, src_id]
That value can be either `true` or `false`, we never set an entry to false though.
I agree checking `connected` would make things a bit cleared but it's redundant.